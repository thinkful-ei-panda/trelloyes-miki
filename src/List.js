import React from 'react';
import Card from './Card';
// import './List.css';

function List(props) {
    return (
        <section className="List">
            <header className="List-header">
            <h2>{props.header}</h2>
            </header>
            <div className="List-cards">
                {props.cards.map((card, i) => <Card key={i} id={card.id} title={card.title} content={card.content} handleDeleteCard={props.handleDeleteCard}/>                    
                )}
                <button type="button" className="List-add-button" onClick={() => props.handleAddCard(props.header)}>
                + Add Random Card
                </button>
            </div>
        </section>
    );
};

export default List;