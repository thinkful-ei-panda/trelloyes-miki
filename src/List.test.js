import React from 'react';
import ReactDOM from 'react-dom';
import List from './List';
import renderer from 'react-test-renderer';
import STORE from './store';

describe('List component', () => {
    // Smoke testing
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<List key={1} header={'Hi'} cards={['a', 'b']} />, div);
        ReactDOM.unmountComponentAtNode(div)
    });

    // Snapshot testing
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<List  key={1} header={'Hi'} cards={['a', 'b']} />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    });

});