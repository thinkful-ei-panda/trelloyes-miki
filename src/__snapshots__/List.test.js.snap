// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      Hi
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
      >
        - Delete Card
      </button>
      <h3 />
      <p />
    </div>
    <div
      className="Card"
    >
      <button
        onClick={[Function]}
      >
        - Delete Card
      </button>
      <h3 />
      <p />
    </div>
    <button
      className="List-add-button"
      onClick={[Function]}
      type="button"
    >
      + Add Random Card
    </button>
  </div>
</section>
`;
